### ISSUE TITLE
fix for Search of AGS layers

### ISSUE THREAD

**Issue by brentfraser** [geomoose/gm3#536]
Not sure about query_params part of my fix:
```
diff --git a/src/gm3/components/map.js b/src/gm3/components/map.js
index e3f98cb..396a95a 100644
--- a/src/gm3/components/map.js
+++ b/src/gm3/components/map.js
@@ -438,7 +438,7 @@ class Map extends React.Component {
                 return name + ' like ' + fix_like(value);
             },
             'ilike': function(name, value) {
-                return name + ' like upper(' + fix_like + ')'
+                return name + " like upper('" + fix_like(value) + "')";
             },
             'eq': function(name, value) {
                 return simple_op('=', name, value);
@@ -485,7 +485,7 @@ class Map extends React.Component {
             // setup the spatial filter.
             query_params.geometryType = geom_type_lookup[queryGeometry.type];
             query_params.geometry = esri_format.writeGeometry(ol_geom);
-            query_params.spatialRel = 'esriSpatialRelIntersects';
+            query_params.spatialRel = 'esriSpatialRelEnvelopeIntersects';//'esriSpatialRelIntersects';
         }

         // build the filter fields.
@@ -495,11 +495,13 @@ class Map extends React.Component {
         }

         const where_str = where_statements.join(' and ');
-
+/*
         query_params.layers = JSON.stringify([{
             layerId: layer_name,
             'where': where_str,
         }]);
+*/
+        query_params.where = where_str;

         // get the query service url.
         const query_url = map_source.urls[0] + '/query/';
@@ -574,8 +576,8 @@ class Map extends React.Component {
         const result_features = [];

         const selection = query.selection;
-        if(selection && selection.geometry && selection.geometry.type === 'Point') {
-            const coords = selection.geometry.coordinates;
+        if(selection && selection[0].geometry && selection[0].geometry.type === 'Point') {
+            const coords = selection[0].geometry.coordinates;
             src.forEachFeatureAtCoordinateDirect(coords, (feature) => {
                 result_features.push(format.writeFeatureObject(feature));
             });
```

testing with 

```
    <map-source name="runways" type="ags-vector">
        <url>https://gis2.co.dakota.mn.us/arcgis/rest/services/DCGIS_OL_Transportation/MapServer/21</url>
        <layer name="runways" selectable="true" title="Runways">
            <template name="identify" auto="true" />
            <style><![CDATA[
            {
                "line-color" : "#010138",
                "line-width" : 2
            }
            ]]></style>
            <template name="search"><![CDATA[
                <div class="search-result">
                    <div class="search-label">
                        {{ properties.Name }}
                    </div>
                    <div class="search-action">
                        <div style="padding: 2px">
                            <a onClick="app.zoomToExtent([{{ properties.boundedBy | join }}], 'EPSG:3857')" class="zoomto-link">
                                <i class="fa fa-search"></i>
                            </a>
                        </div>
                    </div>
                </div>
            ]]></template>
            <template name="select" alias="search"/>
            <template name="gridColumns"><![CDATA[
            [
                {
                    "title": ""
                },
                {
                    "title": "Name",
                    "property" : "Name",
                    "filter" : {
                        "type" : "list"
                    }
                }
            ]
            ]]></template>
            <template name="gridRow"><![CDATA[
            <tr
              onmouseenter="app.highlightFeatures({'OBJECTID' : '{{ properties.OBJECTID }}'}, true)"
              onmouseleave="app.clearHighlight()"
            >
                <td>
                  <a onClick="app.zoomToExtent([{{ properties.boundedBy | join }}], 'EPSG:3857')" class="zoomto-link">
                    <i class="fa fa-search"></i>
                  </a>
                </td>
                <td>
                  {{ properties.Name }}
                </td>
            </tr>
            ]]></template>
        </layer>
    </map-source>
```
and a layer and tool def of:
`           <layer title="AGS Dakota County Runways" src="runways/runways"></layer>`
and
`            <tool name="search-runways" title="Search Runways" type="service"/>`
and in the app.js:
```
    app.registerService('search-runways', SearchService, {
        fields: [
            {type: 'text', label: 'Name', name: 'Name'},
        ],
        searchLayers: ['runways/runways'],
        validateFieldValues: function (fields) {
            let nonEmpty = 0;
            const validateFieldValuesResult = {
                valid: true,
                message: null
            };

            if (fields['Name'] !== undefined && fields['Name'] !== '') {
                    nonEmpty++;
            }

            if (nonEmpty === 0) {
                validateFieldValuesResult.valid = false;
                validateFieldValuesResult.message = 'Please complete at least one field.'
            }
            return validateFieldValuesResult;
        }
    });
```

**Comment by brentfraser** [comment]
![image](https://user-images.githubusercontent.com/1638313/83836478-ca7f0b80-a6b0-11ea-83f2-55e39c4fcf7b.png)

**Comment by brentfraser** [comment]
see PR #539

**Comment by brentfraser** [comment]
Fixed in v3.6.3-beta
